docs/architecture.md

```diff
@@ -0,0 +1,21 @@
+# Profiler Architecture
+
+This document provides an overall look at the architecture of the profiler, and how all the various pieces fit together.
+
+## Gecko Profiler
+
+![Gecko Profiler diagram](images/gecko-profiler.svg)
+
+## Browser
+
+![Browser diagram](images/browser.svg)
```

Comment:
`Private WebExtension API`: What makes this API private? Isn't it just a permission that anybody could flip?